src/Microsoft.Framework.Runtime/ApplicationHostContext.cs

```diff
@@ -173,5 +176,35 @@ public IServiceProvider ServiceProvider
         public string ProjectDirectory { get; private set; }
 
         public string PackagesDirectory { get; private set; }
+
+
+
+        public IEnumerable<ICompilationMessage> LockFileDiagnostics
```

Comment:
Here is the output when you don't have a individual property for lock file diagnostics:
![capture](https://cloud.githubusercontent.com/assets/1383883/8737500/9c011be4-2bd7-11e5-8005-dddaccc3c9a4.PNG)

Note that it says
```
System.InvalidOperationException: Failed to resolve the following dependencies for target framework 'DNX,Version=v4.5.1':
   Newtonsoft.Json 6.0.0
```
but it also says
```
C:\AspNet\Home\samples\latest\ConsoleApp\project.json(3,28): error: The dependency Newtonsoft.Json >= 6.0.0 could not be resolved.
```
So I am trying to get rid of the redundancy here.